src/PeerConnection.ts

```diff
@@ -0,0 +1,82 @@
+import { Duplex } from 'stream'
+import noise from 'noise-peer'
+import multiplex, { MultiplexedStream, SubStream } from 'multiplex'
+import MessageChannel from './MessageChannel'
+import { NetworkMsg } from './NetworkMsg'
+import pump from 'pump'
+
+export interface SocketInfo {
+  type: 'tcp' | 'utp'
+  isClient: boolean
+}
+
+export default class PeerConnection {
+  networkChannel: MessageChannel<NetworkMsg>
```

Comment:
If the multiplex streams are also called channels, I see your affinity for `MessageBus` here: `networkBus`.

Reply:
or even `networkMessageBus`